### PR TITLE
Static site get common header, login, logout working.

### PR DESCRIPTION
**Pull request by doug-lovett**
Issue #: /bcgov/entity#9820

Description of changes:
- Changes to nuxt config to work with sbc common header and login/logout
- Changes to work with nuxt static site generation and firebase deployment
- Update breadcrumb configuration
- Todo: figure out how to configure keycloak.json for different firebase environments. 
- Add tests when solution is stable. Works locally, redirect uri error in dev.

Updated project local/dev env vars to:
BCROS_BASE_URL="http://localhost:8080/"
# BCROS_BASE_URL="https://bcregistries-daxiom.web.app/"
AUTH_URL="https://dev.bcregistry.ca/business/auth/"
NAME_REQUEST_URL="https://dev.bcregistry.ca/namerequest/"
BUSINESS_URL="https://dev.bcregistry.ca/business/auth/home/maintain-business"
COLIN_URL="https://www.corporateonline.gov.bc.ca"
ONESTOP_URL="http://www.bcbusinessregistry.ca/"
SOCIETIES_URL="https://dev.bcregistry.ca/societies/"
PPR_URL="https://ppr-ui-dev.apps.silver.devops.gov.bc.ca/ppr"
WILLS_URL="https://ppr-ui-dev.apps.silver.devops.gov.bc.ca/ppr"
BCOL_URL="https://d1.bconline.gov.bc.ca"
REGISTRY_ACCOUNT_CREATE_URL="https://dev.bcregistry.ca/business/auth/choose-authentication-method"
KEYCLOAK_CONFIG_PATH="keycloak.json"
AUTH_API_URL="https://auth-api-dev.apps.silver.devops.gov.bc.ca/api/v1/"
STATUS_API_URL="https://status-api-dev.apps.silver.devops.gov.bc.ca/api/v1/"

Signed-off-by: Doug Lovett <doug@diamante.ca>

### CODE REVIEW

File: static/keycloak.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "realm": "fcf0kpqr",
+    "auth-server-url": "https://dev.oidc.gov.bc.ca/auth/" ,
+    "ssl-required": "external",
+    "resource": "ppr-web",
+    "public-client": true,
+    "confidential-port": 0
+}
```
Comment (kialj876) [inline static/keycloak.json:8]: Do you need to keep these keycloak files? I think the dev proxy gets it from ppr-dev right now anyways and we will have it in a config mounted in openshift for dev/test/prod

**Review by kialj876 — APPROVED** [review]
Hey Doug this looks good glad you got it all working!

**Review by jordiwes — APPROVED** [review]
LGTM

File: pages/dashboard.vue [hunk 2]
```diff
@@ -40,7 +40,15 @@
   </div>
 </template>
 <script>
-export default {}
+import { SessionStorageKeys } from 'sbc-common-components/src/util/constants'
+
+export default {
+  asyncData(context) {
+    if (!sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)) {
+      context.redirect('/signin')
```
Comment (kialj876) [inline pages/dashboard.vue:48]: Did you try adding this redirect in the router similar to PPR? That way it'll do it for every page in the app. Might not work in nuxt, but worth a try I think!

File: static/keycloak.json (hunk 1 above)
Comment (doug-lovett) [reply static/keycloak.json]: We are currently generating a static site that is deployed to firebase. We run npm run generate, then deploy with a firebase api. So we need all configuration before generating. With this deployment we need to figure out how to update keycloak.json by environment as part of the generate step.

File: pages/dashboard.vue (hunk 2 above)
Comment (doug-lovett) [reply pages/dashboard.vue:48]: Not all of the site is protected right now. I think only the dashboard is. And Nuxt handles routes differently (automatically generated). It's possible to extend or add dynamic routes, I'm still looking at this and need to test with the generated site.

**Pull request merged**